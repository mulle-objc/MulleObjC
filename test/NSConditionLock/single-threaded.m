//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjC/MulleObjC.h>
#else
# import <Foundation/Foundation.h>
# pragma message( "Apple Foundation")
#endif


static void   print_fail_if( int state)
{
   if( state)
      printf( "FAIL\n");
}


int   main( int argc, const char * argv[])
{
   NSConditionLock   *lock;

#ifdef __MULLE_OBJC__
   if( mulle_objc_global_check_universe( __MULLE_OBJC_UNIVERSENAME__) != mulle_objc_universe_is_ok)
      return( 1);
#endif

   lock = [[[NSConditionLock alloc] initWithCondition:1848] autorelease];

   printf( "1. %ld\n", [lock condition]);
   // exercise NSLocking protocol, which is done by NSCondition
   // really
   if( [lock tryLockWhenCondition:1849])
   {
      printf( "succeeded with wrong condition\n");
      return( 1);
   }
   if( ! [lock tryLockWhenCondition:1848])
   {
      printf( "failed with right condition\n");
      return( 1);
   }
   printf( "2. %ld\n", [lock condition]);

   [lock unlockWithCondition:1849];
   printf( "3. %ld\n", [lock condition]);

   [lock lockWhenCondition:1849];
   printf( "4. %ld\n", [lock condition]);
   [lock unlockWithCondition:1850];
   printf( "5. %ld\n", [lock condition]);

   return( 0);
}
