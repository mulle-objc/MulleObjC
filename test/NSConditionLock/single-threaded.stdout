1. 1848
2. 1848
3. 1849
4. 1849
5. 1850
